src/frontend/src/pages/FinancePage/ReimbursementRequestForm/ReimbursementFormView.tsx

```diff
@@ -151,17 +155,26 @@ const ReimbursementRequestFormView: React.FC<ReimbursementRequestFormViewProps>
               <Controller
                 name="vendorId"
                 control={control}
-                render={({ field: { onChange, value } }) => (
-                  <Select onChange={(newValue) => onChange(newValue.target.value)} value={value} error={!!errors.vendorId}>
-                    {allVendors
-                      .sort((a, b) => a.name.localeCompare(b.name))
-                      .map((vendor) => (
-                        <MenuItem key={vendor.vendorId} value={vendor.vendorId}>
-                          {vendor.name}
-                        </MenuItem>
-                      ))}
-                  </Select>
-                )}
+                render={({ field: { onChange, value } }) => {
+                  const mappedVendors = allVendors.sort((a, b) => a.name.localeCompare(b.name)).map(vendorsToAutocomplete);
+                  const onClear = () => {
+                    setValue('vendorId', '');
+                    onChange('');
+                  };
+
+                  return (
+                    <NERAutocomplete
+                      id={'vendor'}
+                      size="medium"
+                      options={mappedVendors}
+                      value={mappedVendors.find((vendor) => vendor.id === value)}
```

Comment:
you do actually need the null, when it returns undefined, the frontend crashes with an error, so I had to add that in for account code a while ago